magda-web-client/src/Components/Dataset/Add/ErrorMessageBox.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { FunctionComponent } from "react";
+import { connect } from "react-redux";
+import { config } from "config";
+import ServerError from "./Errors/ServerError";
+
+type Props = {
+    error?: Error | null;
+};
+
+const ErrorMessageBox: FunctionComponent<Props> = props => {
+    const { error } = props;
+    if (!error) return null;
+    if (error instanceof ServerError) {
+        switch (error.statusCode) {
+            case 500:
+                return (
+                    <div className="au-body au-page-alerts au-page-alerts--error">
+                        <div>
+                            <span>
+                                Magda has encountered an error when submitting
+                                your dataset.
+                            </span>
+                        </div>
+                        <div>
+                            <span>
+                                Everything is still saved locally - please try
+                                again in 5 minutes.
+                            </span>
+                        </div>
+                    </div>
+                );
+            case 400:
+                return (
+                    <div className="au-body au-page-alerts au-page-alerts--error">
+                        <div>
+                            <span>
+                                Magda has encountered the following error:
+                                {error.message ? error.message : "" + error}.
+                            </span>
+                        </div>
+                        <div>
+                            <span>
+                                We've logged this internally, please contact{" "}
+                                <a
+                                    href={`mailto:${
+                                        config.defaultContactEmail
+                                    }`}
+                                >
+                                    {config.defaultContactEmail}
+                                </a>{" "}
+                                for help.
+                            </span>
+                        </div>
+                    </div>
+                );
+            case 403:
```

Comment:
Looks like this needs to cover `401` too:

<img width="899" alt="image" src="https://user-images.githubusercontent.com/900555/63488908-e6376d80-c4f3-11e9-8342-0e09528f9ad2.png">

<img width="430" alt="image" src="https://user-images.githubusercontent.com/900555/63488914-ed5e7b80-c4f3-11e9-955b-b671c8ad038a.png">

Reply:
This probably caused by the other problem below.
But we could hit into a 401 authorisation error in future --- do we need a different error message?

Reply:
Nah I think it should show the 403 message for 401 too :)

Reply:
... although the message should probably be something along the lines of 

> You don't have permission to publish the dataset - this is most likely because you've been logged out - please Sign In again to submit the dataset. Everything has been saved locally.